src/shogun/machine/EnsembleMachine.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+#ifndef __ENSEMBLEMACHINE_H
+#define __ENSEMBLEMACHINE_H
+
+#include <algorithm>
+#include <memory>
+#include <shogun/ensemble/CombinationRule.h>
+#include <shogun/features/Features.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/util/traits.h>
+#include <shogun/util/zip_iterator.h>
+#include <thread>
+#include <vector>
+namespace shogun
+{
+
+	class EnsembleMachine : public Machine
+	{
+
+	public:
+		EnsembleMachine() = default;
+
+		template <
+		    typename T,
+		    typename = std::enable_if_t<traits::is_container<T>::value>>
+		EnsembleMachine(const T& machines)
+		{
+			std::copy(
+			    machines.begin(), machines.end(),
+			    std::back_inserter(m_machines));
+		}
+
+		~EnsembleMachine() = default;
+
+		void init()
+		{
+			SG_ADD(&m_machines, "machine", "Array of machines.", ParameterProperties::HYPER);
+			SG_ADD(&m_combination_rule, "combination_rule", "Combination rule", ParameterProperties::HYPER);
```

Comment:
you register the parameters in a function that seems to be never called. registration should happen when you construct an object, so you should move this to the constructor